sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/UserIdentity.java

```diff
@@ -78,6 +84,24 @@ public String getEmail() {
     return email;
   }
 
+  /**
+   * Return true if groups should be synchronized for this user.
+   *
+   * @since 5.5
+   */
+  public boolean shouldSyncGroups() {
```

Comment:
In fact I think it's a better design to check this property (but name can be changed) than testing that the list of group is null => Because a list should not return null, it's not a good practice.